### ISSUE TITLE
FST transducer support

### ISSUE THREAD

**Issue by eharcevs** [rust-lang/regex#1087]
#### Describe your feature request

I'm trying to upgrade from `regex-automata 0.1.9` and have noticed that newer versions don't have a `transducer` feature anymore, which allowed using a regex `DFA` as an FST `Automaton`. 
Is there a plan to bing this feature back? If not what could be the workaround for using `regex` for `fst`?
Thanks in advance

**Comment by BurntSushi** [comment]
Yes, I just haven't gotten around to it yet. Note that you can always write the trait impls yourself for a wrapper type, so you can unblock yourself if necessary. It could be some time before I get to it.

I will probably add the impls in a new crate or as an optional feature on `fst` itself. I haven't decided yet.

**Comment by eharcevs** [comment]
Got it, thanks!

**Comment by BurntSushi** [comment]
We can leave this open to track when it's done. Thanks!

**Comment by tesioai** [comment]
please Mr Sushi, deliver us from version 0.1.10

**Comment by tisonkun** [comment]
FYI I implement it in the wrapper pattern at - https://github.com/GreptimeTeam/greptimedb/pull/3575

I feel that the implementation is possible to contribute back upstream so that we get it maintained and evolved with the upstream, as well as porting it back can possibly discover some issues that should be fixed.

I have some time to make a patch, but here first to ask if it's the way to go.

**Comment by BurntSushi** [comment]
Sorry, but I'm not going to have time to review this any time soon unfortunately. Submitting a patch probably isn't fruitful at this time, because it's going to require some guidance from me probably in how the crates should be structured. For example, I don't think I want to take a dependency on `fst` inside of `regex-automata`. I think I want to flip that around so that `fst` has an optional dependency on `regex-automata`. And the same for `aho-corasick`: https://github.com/BurntSushi/aho-corasick/blob/56256dca1bcd2365fd1dc987c1c06195429a2e2c/Cargo.toml#L32-L48

**Comment by tisonkun** [comment]
Thanks for your feedback! I won't over commit it so I'll stop here. If I happen to have time make the change on the fst crate, I'll comment there.

Then it seems we can close this issue as won't do in this crate and perhaps create a mirror in fst?

**Comment by BurntSushi** [comment]
I think I'm fine leaving this issue open. If you want to open one on `fst` too, that's fine.
